src/main/java/com/adobe/target/edge/client/service/DefaultTargetService.java

```diff
@@ -53,12 +58,28 @@ public DefaultTargetService(ClientConfig clientConfig, TelemetryService telemetr
     } else {
       this.targetHttpClient = targetHttpClient;
     }
-    this.targetHttpClient.addDefaultHeader(SDK_USER_KEY, SDK_USER_VALUE);
-    this.targetHttpClient.addDefaultHeader(SDK_VERSION_KEY, SDK_VERSION);
+
+    loadDefaultProperties();
+
     this.clientConfig = clientConfig;
     this.telemetryService = telemetryService;
   }
 
+  private void loadDefaultProperties() {
+    Properties defaultProps = new Properties();
+    try {
+      defaultProps.load(getClass().getResourceAsStream("/gradle.properties"));
+    } catch (IOException e) {
+      logger.warn("Unable to load default SDK properties");
+    }
+
+    this.SDK_VERSION = defaultProps.getProperty("version");
+    this.targetHttpClient.addDefaultHeader(SDK_USER_KEY, SDK_USER_VALUE);
+    if (this.SDK_VERSION != null) {
+      this.targetHttpClient.addDefaultHeader(SDK_VERSION_KEY, SDK_VERSION);
+    }
```

Comment:
`loadDefaultProperties` should only be responsible for loading default properties, this other could should be extracted to a separate `setDefaultHeaders` method